.github/workflows/ci.yml

```diff
@@ -222,6 +222,41 @@ jobs:
         env:
           OZONE_WITH_COVERAGE: false
           DEVELOCITY_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  javadoc:
+    needs:
+      - build-info
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.build-info.outputs.sha }}
+      - name: Cache for maven dependencies
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/ozone
+          key: maven-repo-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-repo-
+      - name: Download Ratis repo
+        if: ${{ inputs.ratis_args != '' }}
+        uses: actions/download-artifact@v4
+        with:
+          name: ratis-jars
+          path: |
+            ~/.m2/repository/org/apache/ratis
+      - name: Setup java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 11
+      - name: Run a full build with javadocs generation
+        run: hadoop-ozone/dev-support/checks/build.sh -Pdocs -Dmaven.javadoc.skip=false -Dskip.npx -Dskip.installnpx
+        env:
+          DEVELOCITY_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
```

Comment:
The problem here is that when we compile on Java 8 some unexpected warnings appear, like https://github.com/Daniilchik/ozone/actions/runs/11087840906/job/30809312296#step:8:12947
It only appears on Java 8 and actually there's a lot more problems with Javadoc generation on this version, for example we can't ```@link``` to inner class, see: https://stackoverflow.com/a/50882307
So that's why if we need to ```failOnWarnings``` it’s better to generate Javadoc using a different version of Java rather than version 8.

Reply:
`compile` check runs with Java 11, 17, 21, and with 8 on Mac.  We can probably set `failOnWarnings` conditionally (something like `-Dmaven.javadoc.failOnWarnings=${{  matrix.java != 8 }}`).

Reply:
Thanks for the advice, done.